src/serializers/resource.js

```diff
@@ -141,7 +158,35 @@ function ResourceSerializer(Implementation, model, records, integrator, opts, me
       });
     }
 
-    var serializationOptions = {
+    function defineRelationshipId(records, fields, goDeeper = true) {
+      const recordArray = _.isArray(records) ? records : [records];
+      recordArray.forEach((record) => {
+        fields.forEach((field) => {
+          if (!field.reference) {
+            return;
+          }
+
+          const fieldName = field.field;
+          const [referenceType,referenceKey] = field.reference.split('.');
+          const referenceSchema = Schemas.schemas[referenceType];
+          if (goDeeper && record[fieldName]) {
```

Comment:
We only do it on one level. But we could do it recursively like on forest-rails (@arnaudbesnier Your call)